lms/djangoapps/discussion/static/discussion/js/discussion_board_factory.js

```diff
@@ -61,7 +61,7 @@
 
                 // Set up a router to manage the page's history
                 router = new DiscussionRouter({
-                    courseId: options.courseId,
+                    rootUrl: options.root_url,
```

Comment:
The factory receives `root_url` from the Mako template because it is Python-ese. The factory then passes it on everywhere else in CamelCase. This is confusing (and I had a bug with the logic at one point due to this naming) but I wanted to be consistent in the Mako template and didn't want to change all the parameters. If you think it is worth it then I'll go ahead and change all the Mako parameters to use JS conventions.